java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/sourcenode/SourceRequestHandler.java

```diff
@@ -263,6 +263,154 @@ private void validateDbProperties(
+                            }
+                        }
+                    }
+                    // check whether user has select privilege on table for initial snapshot
+                    try (var stmt =
+                            conn.prepareStatement(
+                                    sqlStmts.getProperty("postgres.table_privilege.check"))) {
+                        stmt.setString(1, props.get(DbzConnectorConfig.TABLE_NAME));
+                        stmt.setString(2, props.get(DbzConnectorConfig.USER));
+                        var res = stmt.executeQuery();
+                        while (res.next()) {
+                            if (!res.getBoolean(1)) {
+                                throw new StatusException(
+                                        Status.INTERNAL.withDescription(
+                                                "Postgres user must have select privilege on table "
+                                                        + props.get(
+                                                                DbzConnectorConfig.TABLE_NAME)));
+                            }
+                        }
+                    }
+                    // check whether publication exists
+                    boolean publicationExists = false;
+                    boolean partialPublication = false;
+                    try (var stmt = conn.createStatement()) {
+                        var res =
+                                stmt.executeQuery(
+                                        sqlStmts.getProperty("postgres.publication_att_exists"));
+                        while (res.next()) {
+                            partialPublication = res.getBoolean(1);
+                        }
+                    }
+                    // pg 15 and up supports partial publication of table
+                    // check whether publication covers all columns
+                    if (partialPublication) {
+                        try (var stmt =
+                                conn.prepareStatement(
+                                        sqlStmts.getProperty("postgres.publication_att"))) {
+                            stmt.setString(1, props.get(DbzConnectorConfig.PG_SCHEMA_NAME));
+                            stmt.setString(2, props.get(DbzConnectorConfig.TABLE_NAME));
+                            var res = stmt.executeQuery();
+                            while (res.next()) {
+                                String[] columnsPub =
+                                        (String[]) res.getArray("attnames").getArray();
+                                var sourceSchema = validate.getTableSchema();
+                                for (int i = 0; i < sourceSchema.getColumnsCount(); i++) {
+                                    String columnName = sourceSchema.getColumns(i).getName();
+                                    if (Arrays.stream(columnsPub).noneMatch(columnName::equals)) {
+                                        break;
```

Comment:
We only use the publication named `dbz_publication` for connectors. So you should check whether the `pubname` is `dbz_publication`.
In this branch, the pub exists but does not cover all columns. We return an error to let the user add `columns` to the publication.